### PR TITLE
unar: add page

### LINKED ISSUE #997: add unar command
unar command

### PR DESCRIPTION
**Pull request by assemusr**
fixes #997

### CODE REVIEW

**Comment by waldyrious** [comment]
Thanks for the contributions, you're on a roll! But please base your branches off `master`, otherwise they include commits from your other PRs. Could you rebase?

**Comment by assemusr** [comment]
yeah I understand what you're saying, but I can't figure out a solution yet.

Here's what I did from the beginning: 
I solved 4 issues, created a branch for each of them rebased off the last one, the 'fuck' branch was the first one I solved, so I think it's the only one based off master (as far as I understand). Which means 'install' based off 'fuck', and 'unar' based off 'install', and 'perl' the last one based off 'unar'. 
Now I merged perl (which has the whole changes I made) to my master fork, I don't know if that would help or not, but I think that I got lost in the documentations.

Sorry about that :D I think it'll take me a while to understand the branching mechanisms.

**Comment by agnivade** [comment]
No worries ! You can always ask for help. :) Let me clarify your most immediate mistake -

>  rebased off the last one.

That was the error. You should create a new branch from master for every PR.

**Some unsolicited advice:**

First of all, you need not use the command line at all. Everything can be done from the UI itself. Often, users find it much easier to use.

Now, if you want to the command line, have the master branch of your local repo always [synced](https://help.github.com/articles/syncing-a-fork/) with the remote upstream. Then for every PR, create a branch from master. Create a commit with the changes, push it. After you get review comments, create more commits to incorporate the changes, push the branch again. It will update that PR. Once, everyone is happy, your PR is merged ! Then, delete your local and remote branch. And don't forget to re-sync your local master before you start work on your next PR !

Some references which I found helpful - https://git-scm.com/docs/gittutorial (both part 1 and 2)

**Comment by waldyrious** [comment]
Actually, the commits don't need to be deleted at all, you just need to rebase the branches onto master. From the [git documentation:](https://git-scm.com/docs/git-rebase) (slightly adapted):

> Let’s assume your `topic` branch is based on branch `next`:

>```
    o---o---o  master
             \
              o---o---o---o---o  next
                               \
                                o---o---o  topic
```
> We want to make `topic` forked from branch `master`; we want our tree to look like this:

>```
    o---o---o  master
            |\
            | o'--o'--o'  topic
             \
              o---o---o---o---o  next
```
> We can get this using the following command:

> `git rebase --onto master next topic`

**Comment by agnivade** [comment]
> that's not what the interactive rebase would do anyway

Umm .. once you open the interactive rebase page, you can remove the lines corresponding to the commit you want to remove. It literally says this in the editor - `If you remove a line here THAT COMMIT WILL BE LOST.`

> git rebase --onto master next topic

Oh wow .. that is handy !

**Comment by waldyrious** [comment]
Yeh, that's confusing, but that language is needed for the general case. Commits that aren't pointed to by any branch reference will be unreachable (but not completely unrecoverable). In this case, if you did an interactive rebase of `topic` onto `master` and removed the lines corresponding to commits D...H, the result would be the same as using the `--onto` example, and since the `next` branch would remain pointing to commit H, it would still be reachable, as well as its ancestors. However, if `next` didn't exist, such a rebase (either with `-i` or with `--onto`) would render commits D...H unreachable. Hope that was clear :)

**Comment by assemusr** [comment]
This one is rebased.
Thanks, for the explanation :) 
It helped, but it's really hard to get your head around :D

**Comment by waldyrious** [comment]
@assemusr for building a visual intuition about git operations, I always recommend the talk [Git for ages 4 and up](https://www.youtube.com/watch?v=1ffBJ4sVUb4), which for me was the "aha!" moment in building a mental model of what a git repository is and how it works :) take a look when you get the time!

File: pages/common/unar.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# unar
+
+> Extract archive file contents.
+
+- Extract archive to current directory:
+
+`unar {{archive}}`
+
+- Extract archive to the specified directory:
```
Comment (waldyrious) [inline pages/common/unar.md]: *an* archive

File: pages/common/unar.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+`unar -o {{path/to/directory}} {{archive}}`
+
+- Force overwrite when file to be unpacked already exist:
```
Comment (waldyrious) [inline pages/common/unar.md]: when --> if; exist --> exists (or file --> files).

Comment (waldyrious) [inline pages/common/unar.md]: (same for examples below)

Comment (assemusr) [inline pages/common/unar.md]: (exist --> exists) has a problem with the inf. tense bot, that's why I omit the 's' in the first place. So, it's better to be (file --> files)

**Comment by waldyrious** [comment]
LGTM 👍

**Comment by agnivade** [comment]
:+1:

**Pull request merged**